aws/cloudformation/cloud_formation_stack.yml.erb

```diff
@@ -1,20 +1,80 @@
 <%
 require 'cdo/aws/cloudfront'
+unless system("git ls-remote --exit-code origin #{branch} > /dev/null")
+  raise 'Current branch needs to be pushed to GitHub with the same name, otherwise deploy will fail.
+To specify an alternate branch name, run `rake stack:start branch=BRANCH`.'
+end
+
+commit = ENV['COMMIT'] || `git ls-remote origin #{branch}`.split.first
+
+ami = commit[0..4]
 unless dry_run
-  update_bootstrap_script.call
   update_certs.call unless cdn_enabled
+  update_cookbooks.call
+  update_bootstrap_script.call
 end
-commit = '521b9fa4942aded6733afd98bc17079d4c23c457'
-#commit = (ENV['COMMIT'] || `git ls-remote origin #{branch} | cut -f1`.strip)
 
+old_commit = '521b9fa4942aded6733afd98bc17079d4c23c457'
+frontends = %i(production adhoc).include?(rack_env)
 -%>
 ---
 AWSTemplateFormatVersion: 2010-09-09
 Description: AWS CloudFormation Template for Code.org application
 Resources:
   VPC: <%= lambda_fn.call 'LookupStackOutputs', StackName: 'VPC', Nonce: 0 %>
+<% if frontends -%>
```

Comment:
The `frontends` conditional is meant to allow this stack to be continuously-deployed in other CI-managed environments (`staging`, `test`, `levelbuilder`) without creating additional infrastructure for managing a fleet of immutable frontend servers (AMI-creation resources, auto scaling group and scaling policies, etc).

The plan is to eventually consolidate the infrastructure so that other environments may (optionally) deploy a frontend-scaling stack as well, just by switching the value of this conditional.